### ISSUE TITLE
Best approach for using rfigshare with continuous integration

### ISSUE THREAD

**Issue by ethanwhite** [ropensci/rfigshare#103]
We're in the process of trying to run rfigshare code as part of a continuous integration build to allow us to automatically archive forecasts (see https://github.com/weecology/portalPredictions/issues/47). I tried to do this using a personal access token for auth but then realized that this wasn't implemented yet (#97).

Since this is being done with CI we can't handle the authentication interactively. Do you have recommendations for the best approach to handling this?

**Comment by cboettig** [comment]
Yeah, figshare dev team has long prioritized it's institutional partnerships over API development. While this makes sense for a business that needs income, it's been a frustration for package development here, which as you can see is pretty stalled.  (though we probably should rouse ourselves to at least fixing those issues we can address).  

It would be great to do this with knb & dataone package instead, which has done a much better job on issues of discovery, redundancy, and provenance for this kind of data.  The catch there is getting a long-lived PAT credential, the default tokens expire daily.  

I think we should have a new interface for zenodo later in the summer (cc @karthik ), the zenodo model isn't a rich as knb but they are doing lots of things right, with a great and actively developed API, versioned dois, more careful, accurate metadata records going to DataCite, and I strong focus on community engagement, community standards and developers (all true of knb as well).  

However, I believe you can already basically accomplish the same thing with @richfitz's excellent https://github.com/ropenscilabs/datastorr.  This just stores the data on GitHub and uses GitHub releases to version the data; so it's simple and elegant; and you can just go log into zenodo once and flip the integration switch, and then each release GitHub release will automatically trigger a Zenodo import, and give you a new, versioned DOI (with all the versioned DOIs on Zenodo nicely linked together, and a separate DOI that always points to the most recent version, none of which you can do on figshare I believe).

**Comment by sckott** [comment]
Nice summary of the topic @cboettig 

what I can do soon, regardless of whether you use figshare or not Ethan, is at least get the PAT workflow working on at least that branch https://github.com/ropensci/rfigshare/tree/sckott - will get working on that soon

**Comment by ethanwhite** [comment]
@cboettig - thanks for the detailed run down. We looked at Zenodo first and couldn't find the functionality we needed in their API, but it was a colleague who looked so I'll check it myself. Thanks for the pointer to datastorr. I'll give it a look. I'm definitely a big fan of the version DOIs.

@sckott - thanks. Definitely don't rush on our account, I just noticed the branch and thought there was an off chance it might be working already.
